src/main/java/com/floragunn/searchguard/tools/tlstool/tasks/Task.java

```diff
@@ -69,14 +72,25 @@ protected Task(Context ctx) {
 
 	public abstract void run() throws ToolException;
 
-	protected KeyPair generateKeyPair(int keySize) throws ToolException {
+	protected KeyPair generateKeyPair(KeyGenParameters parameters) throws ToolException {
 		try {
-			KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA", ctx.getSecurityProvider());
-			generator.initialize(keySize);
+
+		    KeyPairGenerator generator;
+		    if(parameters.getUseEllipticCurves() != null && parameters.getUseEllipticCurves()) {
+		        log.info("Create {} with EC ({})", parameters.getClass().getSimpleName(), parameters.getEllipticCurve());
```

Comment:
log.info output gets displayed to the user of the tlstool by default. Thus, one should consider if this information is actually meaningful to a normal user. If not, one should lower the log level.

Reply:
agreed